Sources/Helpers/AssociatedCellItemFactory.swift

```diff
@@ -0,0 +1,134 @@
+    
+    
+    /// Returns an array of cell items
+    ///
+    /// - Parameters:
+    ///    - array: an array of objects to create cell items for them
+    public func makeCellItems(array: [U]) -> [CollectionViewCellItem] {
+        var cellItems = [CollectionViewCellItem]()
+        for index in 0..<array.count {
+            let object = array[index]
+            cellItems.append(contentsOf: makeCellItems(object: object, index: index))
+        }
+        return cellItems
+    }
+    
+    /// Returns an instance of `UniversalCollectionViewCellItem` and associates provided handlers with them
+    ///
+    /// - Parameters:
+    ///    - object: an object to create a cell item for it
+    ///    - index: the index of the object in the array
+    public func makeUniversalCellItem(object: U, index: Int) -> CollectionViewCellItem {
+        let cellItem = UniversalCollectionViewCellItem<T>()
+        cellItem.configurationHandler = { [weak self] cell in
+            guard let self = self else {
+                return
+            }
+            guard let configurationHandler = self.cellConfigurationHandler else {
+                fatalError("configurationHandler property for the CellItemFactory should be assigned before")
+            }
+            configurationHandler(object, cell, cellItem)
+        }
+        cellItem.sizeConfigurationHandler = { [weak self] (collectionView, sectionItem) -> CGSize in
+            guard let self = self else {
+                return .zero
+            }
+            guard let sizeConfigurationHandler = self.sizeConfigurationHandler else {
+                fatalError("sizeConfigurationHandler property for the CellItemFactory should be assigned before")
+            }
+            return sizeConfigurationHandler(object, collectionView, sectionItem)
+        }
+        cellItemConfigurationHandler?(index, object, cellItem)
+        return cellItem
+    }
+    
+    /// Returns an instance of `UniversalCollectionViewCellItem` and associates provided handlers with them
+    ///
+    /// - Parameters:
+    ///    - configure: a cell configuration handler
+    ///    - size: a cell size configuration handler
+    public func makeCellItem(configure configurationHandler: @escaping (T) -> Void,
+                             size sizeConfigurationHandler: @escaping (UICollectionView, CollectionViewSectionItem) -> CGSize) -> CollectionViewCellItem {
+        let cellItem = UniversalCollectionViewCellItem<T>()
+        cellItem.configurationHandler = configurationHandler
+        cellItem.sizeConfigurationHandler = sizeConfigurationHandler
+        return cellItem
+    }
+}
+
+extension AssociatedCellItemFactory: CellItemFactory {
+    public func makeCellItems(array: [Any]) -> [CollectionViewCellItem] {
+        if let array = array as? [U] {
+            return makeCellItems(array: array)
+        }
+        return []
+    }
+    
+    public func makeCellItems(object: Any, index: Int) -> [CollectionViewCellItem] {
```

Comment:
Can you replace `Any` type to `U` for all of function inside?
```suggestion
    public func makeCellItems(with object: U, index: Int) -> [CollectionViewCellItem] {
```

Reply:
<image src ="https://user-images.githubusercontent.com/15152385/67558555-032d2e80-f739-11e9-90ca-5717e0d643eb.png"/>

Reply:
> Can you replace Any type to U for all of function inside?

No, implemented protocol `CellItemFactory` has no associated type. This function is bridge (or adapter) between `CellItemFactory` and `AssociatedCellItemFactory`